### PR TITLE
Remove the RESTSever.

### PR DESCRIPTION
**Pull request by BramGruneir**
This remove the REST server from the backend.  
I'll remove it from the UI in a follow up PR.

Please take a look at the top of the README file.  Let me know where I'm wrong about what's been done or being worked on and if there is anything you want me to add to either list.

### CODE REVIEW

**Comment by MBerhault** [comment]
why are we removing this? it's convenient to be able to poke at data through the web interface.
Also, I don't see changes to the web interface. Right now, this is the most prominent thing.

**Comment by BramGruneir** [comment]
As I mentioned in the PR description, I'll remove it from the web UI in a follow up PR.  Just waiting for Matt to land #1333 

As to why we're removing it, with our push towards the structured data, it was felt that there was no need for the REST endpoint.  It's an outdated model for what we're trying to accomplish and was starting to get in the way.
@petermattis can give you a fully explanation.

**Comment by petermattis** [comment]
The current REST api exposes the proto responses directly. That is less than ideal because those responses have internal implementation details. The REST api has no support for transactions removing a huge value proposition for cockroach. And this is likely to all change when the structured data api comes along shortly. Let's trim our surface area to that which we're actively developing and committed to.

File: README.md [hunk 1]
```diff
@@ -19,18 +19,25 @@
 
 ## Status
 
-**ALPHA**
+**Completed**
```
Comment (tamird) [inline README.md]: oh this word scares me

**Comment by tamird** [comment]
LGTM but ping me when the tests are passing

File: README.md (hunk 1 above)
Comment (BramGruneir) [inline README.md]: Alright... suggest something else.  Available?

File: README.md [hunk 2]
```diff
@@ -19,18 +19,25 @@
...
+(last updated June 11th, 2015)
```
Comment (MBerhault) [inline README.md]: this entire section should probably be rolled into the roadmap, with only the ALPHA status on the main README.

**Comment by BramGruneir** [comment]
I'll rebase after your fix #1346 is in.

**Comment by MBerhault** [comment]
LGTM. testrace seems to be running into its time limit and is becoming a bit flaky on circleCI.

File: README.md (hunk 2 above)
Comment (BramGruneir) [inline README.md]: I can do that.  But what is our Alpha status?

Comment (MBerhault) [inline README.md]: our status is "alpha". We already show that in the readme header.

**Comment by bdarnell** [comment]
LGTM

**Comment by petermattis** [comment]
LGTM

File: README.md (hunk 2 above)
Comment (BramGruneir) [inline README.md]: I have no problem pulling all of this section out.  I'll do just that.

**Pull request merged**